scripts/jenkins/cloud/ansible/roles/list_packages/library/diff_yaml.py

```diff
@@ -0,0 +1,129 @@
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+
+
+from __future__ import print_function
+
+from re import findall
+
+from ansible.module_utils.basic import AnsibleModule
+
+from deepdiff import DeepDiff
+
+import yaml
+
+DOCUMENTATION = '''
+---
+module: diff_yaml
+short_description: do a diff of yaml data
+description: |
+  Module will compare 2 given yaml files and return
+  structured diff. All is done on local machine
+author: SUSE Linux GmbH
+options:
+  file1:
+    description: Path to the file
+  file2:
+    description: Path to the file
+  output:
+    description: Name of file to write diff into
+'''
+
+EXAMPLES = '''
+- diff_yaml:
+    file1: path/to/yaml/file
+    file2: path/to/yaml/file
+    output: path/to/diff/file
+  register: _result
+'''
+
+
+def loadFiles(file1, file2):
+    with open(file1) as f:
+        firstFile = yaml.safe_load(f)
+    with open(file2) as f:
+        secondFile = yaml.safe_load(f)
+    return (firstFile, secondFile)
+
+def addToDict(dictItems, valueObj, keyObj, item, action):
+    packageName = keyObj[0]
+    #packageName = action if len(keyObj) == 1 else keyObj[0]
+    attribute = keyObj[0] if len(keyObj) == 1 else keyObj[1]
+    #try:
+    #    attribute = keyObj[1]
+    #except:
+    #    attribute = ke
+    attrValue = valueObj
+    if packageName in dictItems:
+#        dictItems[packageName][attribute] = attrValue
+        dictItems[action][packageName][attribute] = attrValue
+    else:
+        # if dict for packageName is empty initiate it and put first record
+#        dictItems[packageName] = {}
+#        dictItems[packageName][attribute] = attrValue
+        dictItems[action] = {}
+        dictItems[action][packageName] = {}
+        if action != "values_changed":
+            dictItems[action][packageName]
+        else:
+            dictItems[action][packageName][attribute] = attrValue
+
+
+def main():
+    try:
+        dictItems = {}
+
+        argument_spec = dict(
+            file1=dict(type='str', required=True),
+            file2=dict(type='str', required=True),
+            output=dict(type='str', required=True)
+        )
+        module = AnsibleModule(argument_spec=argument_spec,
+                               supports_check_mode=False)
+        file1 = module.params['file1']
+        file2 = module.params['file2']
+        output = module.params['output']
+# temporary variables locally set-up
+#        file1 =  "/tmp/rpms/rpms_compute1_after_deploy.yaml"
+#        file2 =  "/tmp/rpms/rpms_compute1_after_update.yaml"
+#        output = "/tmp/rpms/diffs/rpms_compute1.yaml"
+        (dataMap1, dataMap2) = loadFiles(file1, file2)
+        # do the diff
+        diffik = DeepDiff(dataMap1, dataMap2)
+        for i in diffik:
+            action = i
```

Comment:
`action = i.split("_")[-1]` would give you more generic action names

Reply:
if you decided to include that, remember to update `"values_changed"` above as well.

Reply:
I would prefer to keep it in a current way.